src/main/java/org/cactoos/scalar/AvgOf.java

```diff
@@ -83,16 +83,13 @@ public AvgOf(final Integer... src) {
             }
             return sum / total;
         }, () -> {
-            int sum = 0;
-            int total = 0;
-            for (final int val : src) {
-                sum += val;
+            double sum = 0;
```

Comment:
@allanws You should be explicit with double values. So, use `0.0d` instead of `0`.